apps/cowswap-frontend/src/modules/permit/utils/generatePermitHook.ts

```diff
@@ -0,0 +1,101 @@
+import { MaxUint256 } from '@ethersproject/constants'
+
+import { Eip2612PermitUtils } from '@1inch/permit-signed-approvals-utils'
+
+import { GP_VAULT_RELAYER } from 'legacy/constants'
+
+import { Web3ProviderConnector } from './Web3ProviderConnector'
+
+import { FAKE_SIGNER } from '../const'
+import { PermitHookParams } from '../types'
+
+const cachePrefix = 'permitCache-'
+const pendingRequests: { [permitKey: string]: Promise<string> | undefined } = {}
+
+const FIVE_YEARS_IN_SECONDS = 5 * 365 * 24 * 60 * 60
+
+function getDeadline() {
+  return Math.ceil(Date.now() / 1000) + FIVE_YEARS_IN_SECONDS
+}
+
+export async function generateQuotePermitHook(params: PermitHookParams): Promise<string> {
+  const { inputToken, chainId, permitInfo, provider, account } = params
+  const tokenAddress = inputToken.address
+  const tokenName = inputToken.name || tokenAddress
+  const permitKey = `${cachePrefix}${inputToken.address.toLowerCase()}-${chainId}${account ? `-${account}` : ''}`
```

Comment:
capitalised `cachePrefix` since is a const declared outside of the scope of this function

consider making a private function to get the key

Reply:
I meant to refactor this but forgot.
Will do